src/backend/opengl/CMakeLists.txt

```diff
@@ -1,156 +1,130 @@
-    )
+find_package(OpenGL REQUIRED)
 
-FILE(GLOB api_headers
-    "${PROJECT_SOURCE_DIR}/include/*.h"
-    "${PROJECT_SOURCE_DIR}/include/fg/*.h"
-    )
+set(BackendTarget "forge")
 
-FILE(GLOB c_api_headers
-    "${PROJECT_SOURCE_DIR}/src/api/c/*.hpp"
-    )
-FILE(GLOB c_api_sources
-    "${PROJECT_SOURCE_DIR}/src/api/c/*.cpp"
-    )
-FILE(GLOB cpp_api_headers
-    "${PROJECT_SOURCE_DIR}/src/api/cpp/*.hpp"
-    )
-FILE(GLOB cpp_api_sources
-    "${PROJECT_SOURCE_DIR}/src/api/cpp/*.cpp"
-    )
+add_library(${BackendTarget} SHARED "")
 
-FILE(GLOB bknd_headers
-    "${PROJECT_SOURCE_DIR}/src/backend/*.hpp"
-    )
+set_target_properties(${BackendTarget} PROPERTIES FOLDER Forge)
+set_target_properties(${BackendTarget} PROPERTIES POSITION_INDEPENDENT_CODE ON)
+set_target_properties(${BackendTarget} PROPERTIES CXX_STANDARD 11)
+set_target_properties(${BackendTarget} PROPERTIES LINKER_LANGUAGE CXX)
 
-FILE(GLOB bknd_sources
-    "${PROJECT_SOURCE_DIR}/src/backend/*.cpp"
+target_compile_definitions(${BackendTarget}
+    PRIVATE
+        WIN32_MEAN_AND_LEAN=$<PLATFORM_ID:Windows>
+        OS_WIN=$<PLATFORM_ID:Windows>
+        OS_MAC=$<PLATFORM_ID:Darwin>
+        OS_LNX=$<PLATFORM_ID:Linux>
     )
 
-FILE(GLOB gl_headers
-    "*.hpp"
-    "*.h"
+if(WIN32)
+    target_compile_definitions(${BackendTarget} PUBLIC FGDLL)
+endif(WIN32)
+
+target_sources(${BackendTarget}
+    PRIVATE
+        ${CMAKE_CURRENT_SOURCE_DIR}/backend.hpp
+        ${CMAKE_CURRENT_SOURCE_DIR}/chart_impl.hpp
+        ${CMAKE_CURRENT_SOURCE_DIR}/chart_impl.cpp
+        ${CMAKE_CURRENT_SOURCE_DIR}/colormap_impl.hpp
+        ${CMAKE_CURRENT_SOURCE_DIR}/colormap_impl.cpp
+        ${CMAKE_CURRENT_SOURCE_DIR}/common.hpp
+        ${CMAKE_CURRENT_SOURCE_DIR}/common.cpp
+        ${CMAKE_CURRENT_SOURCE_DIR}/err_opengl.hpp
+        ${CMAKE_CURRENT_SOURCE_DIR}/err_opengl.cpp
+        ${CMAKE_CURRENT_SOURCE_DIR}/font_atlas_impl.hpp
+        ${CMAKE_CURRENT_SOURCE_DIR}/font_atlas_impl.cpp
+        ${CMAKE_CURRENT_SOURCE_DIR}/font_impl.hpp
+        ${CMAKE_CURRENT_SOURCE_DIR}/font_impl.cpp
+        ${CMAKE_CURRENT_SOURCE_DIR}/gl_native_handles.hpp
+        ${CMAKE_CURRENT_SOURCE_DIR}/gl_native_handles.cpp
+        ${CMAKE_CURRENT_SOURCE_DIR}/histogram_impl.hpp
+        ${CMAKE_CURRENT_SOURCE_DIR}/histogram_impl.cpp
+        ${CMAKE_CURRENT_SOURCE_DIR}/image_impl.hpp
+        ${CMAKE_CURRENT_SOURCE_DIR}/image_impl.cpp
+        ${CMAKE_CURRENT_SOURCE_DIR}/plot_impl.hpp
+        ${CMAKE_CURRENT_SOURCE_DIR}/plot_impl.cpp
+        ${CMAKE_CURRENT_SOURCE_DIR}/surface_impl.hpp
+        ${CMAKE_CURRENT_SOURCE_DIR}/surface_impl.cpp
+        ${CMAKE_CURRENT_SOURCE_DIR}/update_buffer.cpp
+        ${CMAKE_CURRENT_SOURCE_DIR}/vector_field_impl.hpp
+        ${CMAKE_CURRENT_SOURCE_DIR}/vector_field_impl.cpp
+        ${CMAKE_CURRENT_SOURCE_DIR}/window_impl.hpp
+        ${CMAKE_CURRENT_SOURCE_DIR}/window_impl.cpp
     )
 
-FILE(GLOB gl_sources
-    "*.c"
-    "*.cpp"
+if(${USE_WINDOW_TOOLKIT} STREQUAL "glfw3")
```

Comment:
USE_WINDOW_TOOLKIT seems like a boolean variable. Perhaps change it to WINDOW_TOOLKIT